### PR TITLE
rtklib: add recipe

### PR DESCRIPTION
**Pull request by valgur**
Specify library name and version:  **rtklib/2.4.3-b34**

Adds a recipe for [RTKLIB](https://github.com/tomojitakasu/RTKLIB): a popular open-source tool for precise GNSS positioning.

[Packaging status](https://repology.org/project/rtklib/versions)

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

TODO

- [x] `stream.c:362:9: error: use of undeclared identifier 'B460800'`

### CODE REVIEW

File: recipes/rtklib/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        "shared": False,
+        "fPIC": True,
+        "trace": False,
+        "enable_glonass": True,
+        "enable_qzss": True,
+        "enable_galileo": True,
+        "enable_beidou": True,
+        "enable_irnss": True,
+        "num_frequencies": 5,
+        "num_ext_obs_codes": 3,
```
[the remark this answers is not in the document]
Comment (valgur) [reply recipes/rtklib/all/conanfile.py:41]: The default values are based on https://github.com/tomojitakasu/RTKLIB/blob/v2.4.3-b34/app/consapp/rnx2rtkp/gcc/makefile#L5
Also, 5 is the correct value anyway for `num_frequencies` after GPS started broadcasting L5 signals in 2014.
But it's a good point. I added a comment mentioning this in the recipe as well now.

File: recipes/rtklib/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    @property
+    def _public_defines(self):
+        # https://github.com/tomojitakasu/RTKLIB/blob/v2.4.3-b34/src/rtklib.h#L6-L15
+        defs = {}
+        defs["ENAGLO"] = bool(self.options.enable_glonass)
+        defs["ENAQZS"] = bool(self.options.enable_qzss)
+        defs["ENAGAL"] = bool(self.options.enable_galileo)
+        defs["ENACMP"] = bool(self.options.enable_beidou)
+        defs["ENAIRN"] = bool(self.options.enable_irnss)
```
[the remark this answers is not in the document]
Comment (valgur) [reply recipes/rtklib/all/conanfile.py]: Simplified the logic a bit. The `defs` dict now expects wither a string or a truthy value.

File: recipes/rtklib/all/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RubenRBS) [reply recipes/rtklib/all/conanfile.py:41]: Thanks for the extra info, really interesting TIL!
And yeah comments in the recipes for things like this make the next one to come around a bit easier :)

**Comment by RubenRBS** [comment]
@valgur this is failing with `shared=True` in macos (I wonder why not onLinux too!) while trying to find the symbols for the externed functions:

```
extern int showmsg(const char *format,...);
extern void settspan(gtime_t ts, gtime_t te);
extern void settime(gtime_t time);
```

Let me know hwo you want to handle this so that we can move forward with it :)

**Comment by valgur** [comment]
Oh, I did not notice that the failure is due to `shared=True` on macOS. Thanks, @RubenRBS! I hope the added linker flag fixes the issue.

**Comment by RubenRBS** [comment]
More shared issues, in Windows this time. Are you able to debug there?

File: recipes/rtklib/all/conandata.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+sources:
+  "2.4.3-b34":
+    url: "https://github.com/tomojitakasu/RTKLIB/archive/refs/tags/v2.4.3-b34.tar.gz"
+    sha256: "a9ce13293ef71b028036cb98a114b9bba96bdd439a0aeebca8e9c73156e461cf"
+patches:
+  "2.4.3-b34":
+    - patch_file: "patches/fix-dll-export-of-global-vars.patch"
+      patch_description: "Fix missing EXPORT attributes on global variables required for shared Windows builds"
```
Comment (RubenRBS) [inline recipes/rtklib/all/conandata.yml:8]: Has this been reported upstream?

**Review by RubenRBS — APPROVED** [review]
Looks good to me

**Review by SSE4 — APPROVED** [review]

**Pull request merged**